src/trace/cold-start-tracer.ts

```diff
@@ -15,28 +14,27 @@ export class ColdStartTracer {
   private tracerWrapper: TracerWrapper;
   private parentSpan?: SpanWrapper;
   private lambdaFunctionName?: string;
-  private coldStartSpanFinishTime: number;
   private minDuration: number;
   private ignoreLibs: string[];
 
   constructor(coldStartTracerConfig: ColdStartTracerConfig) {
     this.tracerWrapper = coldStartTracerConfig.tracerWrapper;
     this.parentSpan = coldStartTracerConfig.parentSpan;
     this.lambdaFunctionName = coldStartTracerConfig.lambdaFunctionName;
-    this.coldStartSpanFinishTime = coldStartTracerConfig.coldStartSpanFinishTime;
     this.minDuration = coldStartTracerConfig.minDuration;
     this.ignoreLibs = coldStartTracerConfig.ignoreLibs.split(",");
   }
 
   trace(rootNodes: RequireNode[]) {
     const coldStartSpanStartTime = rootNodes[0]?.startTime;
-    const coldStartSpan = this.createColdStartSpan(coldStartSpanStartTime, this.parentSpan);
+    const coldStartSpanEndTime = rootNodes[rootNodes.length - 1]?.endTime;
```

Comment:
If a require node were created during customer's code because they dynamically loaded the module,  `rootNodes[rootNodes.length - 1].endTime` can potentially end up outside the outer cold-trace span? (I haven't tried this in NodeJS but python example is like [this](https://ddserverless.datadoghq.com/apm/trace/15416331496885506621?colorBy=service&env=dev&graphType=flamegraph&shouldShowLegend=true&sort=time&spanID=14889763467639317383&spanViewType=metadata&timeHint=1680046997360))

Reply:
That's a good point. Maybe we take the oldest timestamp, either the _last_ root node end time or the parent span start time?